### PR TITLE
[Orc8r] Adding APN swagger changes and skeleton code for servicer and api handlers

### PR DESCRIPTION
**Pull request by karthiksubraveti**
Signed-off-by: Karthik Subraveti <ksubraveti@fb.com>

## Summary
Added swagger changes for gateway_apn_configs
Defined following new REST API endpoints

Screen Shot 2021-06-15 at 9 43 12 PM

## Test Plan
Tested the APIs with skeleton handlers. 

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/protos/apn.proto [hunk 1]
```diff
@@ -80,17 +82,15 @@ message APNConfiguration {
 // Network wide APN configuration
 message APNConfig {
   uint32 id = 1;
-  // Contains either the APN Name or wildcard "*"
-  string service_selection = 2;
```
Comment (ulaskozat) [inline lte/protos/apn.proto:84]: why did you remove the service_selection? This is used to name the APN (e.g., "magma.ipv4", "ims", "internet", etc.)

File: lte/protos/apn.proto [hunk 2]
```diff
@@ -80,17 +82,15 @@ message APNConfiguration {
...
+  QoSProfile qos_profile = 2;
+  AggregatedMaximumBitrate ambr = 3;
   enum PDNType {
       IPV4 = 0;
       IPV6 = 1;
       IPV4V6 = 2;
       IPV4_OR_IPV6 = 3;
   }
-  PDNType pdn = 5;
+  PDNType pdn = 4;
```
Comment (ulaskozat) [inline lte/protos/apn.proto:93]: we do not renumber the legacy fields for backwards compatibility. use deprecated for fields that are no longer used.

File: lte/protos/apn.proto (hunk 1 above)
Comment (karthiksubraveti) [reply lte/protos/apn.proto:84]: Earlier APNConfig was housed within a subscriber and it made sense to name the APN. This is the new config model for APN which is independently streamed down, the service_selection attribute needn't be a part of the APN config. In subscriber config, I added a new attribute
```
message SubscriberAPNConfig {
  uint32 apn_id = 1;
  string assigned_static_ip = 2;
}
```
subscriber_apn_config . This does the apn selection and also provides any additional APN specific subscriber configuration.

File: lte/protos/apn.proto (hunk 2 above)
Comment (karthiksubraveti) [reply lte/protos/apn.proto:93]: This is not yet being used. This is part of the new proto struct I'm adding as a part of the feature. Hence did the renumbering.

File: lte/protos/apn.proto [hunk 3]
```diff
@@ -123,12 +123,12 @@ message GatewayAPNConfig {
     APNFilter apn_filter = 2;
```
Comment (ulaskozat) [inline lte/protos/apn.proto:123]: who generates apn_id? how is it used?

What is the purpose of APNFilter?

Comment (karthiksubraveti) [reply lte/protos/apn.proto:123]: apn_id(Its incorrectly marked as u32, will change this to string) identifies the network wide APN entity. APNFilter is just a future feature, where we can customize the APN specific behavior per gateway on whether we choose to permit or block. I can remove it if you feel its a distraction.

File: lte/protos/apn.proto (hunk 1 above)
Comment (ulaskozat) [reply lte/protos/apn.proto:84]: let's not change the field names such as  service_selection or context_id otherwise we will lose parity on s6a interface in federated vs. non-federated cases.

Comment (ulaskozat) [reply lte/protos/apn.proto:84]: currently subscriberdb (in non-federated case) generates the context_id (uint32), but actually it makes more sense for the orc8r to assign it as well.

**Comment by themarwhal** [comment]
@karthiksubraveti Is this PR still up to date / open? Seeing merge conflicts + check failures.

**Comment by hcgatewood** [comment]
Closing this PR till @karthiksubraveti returns from paternity leave
